src/main/java/ru/yandex/clickhouse/util/ClickHouseHttpClientBuilder.java

```diff
@@ -209,4 +240,31 @@ private KeyStore getKeyStore()
           caInputStream.close();
       }
   }
+
+    private CredentialsProvider getDefaultCredentialsProvider() {
+        if (!isConfigurationValidForAuth(properties)) {
+            return null;
+        }
+        HttpHost targetHost = getTargetHost(properties);
+        CredentialsProvider credsProvider = new BasicCredentialsProvider();
+        credsProvider.setCredentials(
+            new AuthScope(targetHost.getHostName(), targetHost.getPort()),
+            new UsernamePasswordCredentials(
+                properties.getUser() != null  ? properties.getUser() : "default",
+                properties.getPassword() != null ? properties.getPassword() : ""));
+        return credsProvider;
+    }
+
+    private static HttpHost getTargetHost(ClickHouseProperties props) {
+        return new HttpHost(
+            props.getHost(),
+            props.getPort(),
+            props.getSsl() ? "https" : "http");
+    }
+
+    private static boolean isConfigurationValidForAuth(ClickHouseProperties props) {
+        return props.getHost() != null
```

Comment:
I'm not sure why we need to check host here - it's always `null` in `BalancedClickhouseDataSource`. Will this cause authentication issue when connecting to a cluster?

Reply:
Good point. I had not thought about this. Will try to find out how the `BalancedClickhouseDataSource` works.

Reply:
Added some tests for common scenarios to `BalancedClickhouseDataSourceTest`, do you have any others you can think of? The host is set in 'ClickhouseJdbcUrlParser.parseClickhouseUrl` from the URL, so we hopefully do not have to change the implementation of `BalancedClickhouseDataSource`

Reply:
I see. Thanks for the detailed explanation.  I don't have anything to add then.